src/utils/hooks/useFetchProductCategories.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+import { API_BASE_URL } from '../constants';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useFetchProductCategories() {
+  const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
+  const [productCategories, setProductCategories] = useState(() => ({
+    categories: [],
+    isLoading: true,
+  }));
```

Comment:
Me agrada que estas guardando el estado de _cargando_, podemos ir un paso mas y agregar el estado de error, manejar cada estado de manera separada es una buena idea porque es simple la actualización y el hook no es muy grande.

Podemos modificar el código con algo como, al definir nuestro estado inicial podemos controlar el comportamiento que el app tendrá cuando llamemos el hook, por eso `loading` se inicializa en `true`

```suggestion
  const [categories, setCategories] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
```